### PR TITLE
literature: add ``recid`` to authors

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: inspirehep/records/marshmallow/literature/common/author.py [hunk 1]
```diff
@@ -22,6 +23,7 @@ class AuthorSchemaV1(Schema):
     inspire_roles = fields.Raw()
     raw_affilitaions = fields.Raw()
     record = fields.Raw()
+    recid = fields.Method("get_recid", attribute="record", default=missing)
```
Comment (harunurhan) [inline inspirehep/records/marshmallow/literature/common/author.py]: `attribute` is redundant in that case.

File: inspirehep/records/marshmallow/literature/common/author.py [hunk 2]
```diff
@@ -43,6 +45,11 @@ def get_last_name(self, data):
 
         return missing
 
+    def get_recid(self, data):
```
Comment (harunurhan) [inline inspirehep/records/marshmallow/literature/common/author.py]: this didn't exist anywhere else before?

File: tests/unit/records/marshmallow/literature/common/test_author.py [hunk 3]
```diff
@@ -125,3 +125,23 @@ def test_author_es_enchancement_without_last_name():
...
+
+
+def test_author_add_recid():
+    schema = AuthorSchemaV1()
+
+    dump = {
+        "full_name": "Castle, Frank",
+        "record": {"$ref": "http://localhost:5000/api/authors/1"},
+    }
+    expected = {
```
Comment (harunurhan) [inline tests/unit/records/marshmallow/literature/common/test_author.py]: can you assert only `recid`?

File: inspirehep/records/marshmallow/literature/common/author.py (hunk 1 above)
Comment (drjova) [reply inspirehep/records/marshmallow/literature/common/author.py]: you're right 👍

File: inspirehep/records/marshmallow/literature/common/author.py (hunk 2 above)
Comment (drjova) [reply inspirehep/records/marshmallow/literature/common/author.py]: AFAIK nope

**Comment by drjova** [comment]
@harunurhan updated

File: inspirehep/records/marshmallow/literature/common/author.py (hunk 2 above)
Comment (harunurhan) [reply inspirehep/records/marshmallow/literature/common/author.py]: ok I guess we can leave it like that, until we need it elsewhere.

**Comment by drjova** [comment]
Supersede by https://github.com/inspirehep/inspirehep/pull/182

**Pull request closed without merging**